### PR TITLE
lib/fixed-points.nix: add an example for extends

### PR DESCRIPTION
**Pull request by typetetris**
- helped me understand how extends works, hopefully it can help others too

###### Motivation for this change

Just add an example for how extends unfolds, so one can easier
understand how it works.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/fixed-points.nix [hunk 1]
```diff
@@ -41,6 +41,13 @@ rec {
   # think of it as an infix operator `g extends f` that mimics the syntax from
   # Java. It may seem counter-intuitive to have the "base class" as the second
   # argument, but it's nice this way if several uses of `extends` are cascaded.
+  #
```
Comment (roberth) [inline lib/fixed-points.nix]: Needs a bit of introduction like "Let's see how `extends` evaluates the example."

Comment (roberth) [reply lib/fixed-points.nix]: `g` itself should also be introduced as an example, which the pre-existing text fails to do.

Comment (typetetris) [reply lib/fixed-points.nix]: g is defined in the text above the context

**Comment by typetetris** [comment]
I added some introductory text and mentioned, that I use the `g` and `f` as defined in the above comments a few paragraphs above.

**Pull request merged**